### DISCUSSION TITLE
Guess target arch on building time for multi arch builds

### DISCUSSION THREAD

**Discussion by alvarogg777** [containers/buildah#3199] [Q&A]
Hi buildah community!

How can anyone guess the target ARCH on building time, so building with --arch arm64 and --build-arg=ARCH=arm64v8 gets evaluated to an arm64 env variable to later download software from external sources (eg arm64 git releases).

I cannot figure out how to get the right arch I am targetting to. from **uname -m**, using the ENV command.

It seems that the only way out is to build a base image and add some configs to set extra information when the image is built, _besides looking at some /lib files arch extensions._
But I think it may be useful as well to expose arch information as an env variable as docker does, so anyone could refactor some Dockerfiles faster to support multiarch.

**Comment by rhatdan** [comment]
Can you use that `arch` executable and work off of that?
$ arch
x86_64

**Reply by alvarogg777** [reply]
I did, but it doesn't even get listed as env var.. running on v1.20.1

**Reply by rhatdan** [reply]
I am not sure what you expect?

**Reply by alvarogg777** [reply]
I'm kinda new on buildah @rhatdan, but basically I'm running this on a Makefile loop for each arch

> arch_image=$(image_name):$(image_tag)-$$arch ;
> buildah bud -t $$arch_image --arch $$arch . \
> buildah push --authfile $(HOME)/.docker/config.json $$arch_image ; \
> buildah manifest add $(target) $$arch_image ; \

And my Dockerfile contains some RUN commands eg.
curl -L https://storage.googleapis.com/kubernetes-release/release/v${KUSTOMIZE_VER}/bin/linux/$arch/kubectl

To later push the manifest with all archs

**Reply by edsantiago** [reply]
I don't think I have enough context to be able to diagnose (e.g. I don't know what your Makefile looks like); but if you want a simple example of a multi-arch manifest, you could look at: https://github.com/containers/podman/blob/b01ec314f58941c955a7b6f878ce995ab3239656/test/system/build-testimage

**Reply by rhatdan** [reply]
How about

arch_image=$(image_name):$(image_tag)-$$arch ;
buildah bud `--manifest $(target)` -t $$arch_image --arch $$arch .
buildah push --authfile $(HOME)/.docker/config.json $$arch_image ;

**Reply by alvarogg777** [reply]
@edsantiago  @rhatdan Here's my Makefile and Dockerfile.. what I've been trying so far.. 

https://gist.github.com/alvarogg777/53361d125cbe05cc6c70ff0b69acde60/#file-dockerfile

I had to manually grep and parse from libc (musl or glibc) extensions the arch..
